src/MovieList/MovieList.jsx

```diff
@@ -1,45 +1,50 @@
-import PropTypes from 'prop-types';
-import React from 'react';
+import React, { Component } from 'react';
+import { withRouter } from 'react-router';
+
 import MovieAsset from './MovieAsset/MovieAsset';
 import styles from './MovieList.scss';
+import data from '../assets/data.json';
 
-const MovieList = ({ assets, handleShowDetails }) => {
-  const handleShowMovieDetails = (movieDetails) => {
-    handleShowDetails(movieDetails);
-  };
-  return (
-    <div className={styles.movieList}>
-      {assets.map(asset => (
-        <MovieAsset
-          key={asset.show_id}
-          asset={asset}
-          handleShowDetails={handleShowMovieDetails}
-        />
-      ))}
-    </div>
-  );
-};
+class MovieList extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      assets: data,
+    };
+  }
 
-MovieList.propTypes = {
-  handleShowDetails: PropTypes.func,
-  assets: PropTypes.arrayOf(PropTypes.shape({
-    category: PropTypes.string,
-    director: PropTypes.string,
-    mediatype: PropTypes.number,
-    poster: PropTypes.string,
-    rating: PropTypes.string,
-    release_year: PropTypes.string,
-    runtime: PropTypes.string,
-    show_cast: PropTypes.string,
-    show_id: PropTypes.number,
-    show_title: PropTypes.string,
-    summary: PropTypes.string,
-    unit: PropTypes.number,
-  })).isRequired,
-};
+  componentWillMount = () => {
+    if (!(/name=([\w]*[\s]*)*&type=(title|director)/.test(this.props.location.search))) {
+      this.setState({
+        assets: [],
+      });
+    }
+  };
 
-MovieList.defaultProps = {
-  handleShowDetails: () => {},
-};
+  componentWillReceiveProps = (nextProps) => {
+    const { location } = this.props;
+    if (/name=([\w]*[\s]*)*&type=(title|director)/.test(nextProps.location.search)) {
```

Comment:
The same is here. We need to check pathname if it is search. If so, then we check previous pathname and if it is the same then we check location.search. And if it is true then we  don't need to update anything and return.
`if (/search/.test(nextProps.location.pathanme)) {
  if (/search/.test(this.props.location.pathname) && nextProps.location.search === this.props.location.search) {
    return;
  }
}`